app/src/main/java/com/github/mik629/android/fundamentals/di/AppComponent.kt

```diff
@@ -0,0 +1,32 @@
+package com.github.mik629.android.fundamentals.di
+
+import android.app.Application
+import com.github.mik629.android.fundamentals.App
+import com.github.mik629.android.fundamentals.domain.repositories.MoviesRepository
+import com.github.mik629.android.fundamentals.ui.AppActivity
+import com.github.mik629.android.fundamentals.ui.BaseFragment
+import dagger.BindsInstance
+import dagger.Component
+import javax.inject.Singleton
+
+@Singleton
+@Component(modules = [AppModule::class, CommonUiModule::class, DataModule::class, NetworkModule::class, ViewModelsModule::class])
+interface AppComponent {
+    //    fun provideRouter(): Router
+
+    fun provideMoviesRepository(): MoviesRepository
+
+    fun inject(app: App)
+
+    fun inject(appActivity: AppActivity)
+
+    fun inject(fragment: BaseFragment)
```

Comment:
no BaseFragment
as when u will add new fragment with some another deps, u won't see compile error when u will forgot to add it to
AppComponent.inject(fragment: AnotherFragment)
and will get runtime error